text/0000-introduce empty ranges.md

```diff
@@ -0,0 +1,255 @@
+Range(inf, 0, -10): .is_empty() = false, .is_infinite() = true
+Range(inf, 0, -inf): .is_empty() = false, .is_infinite() = true
+Range(-inf, 0, 0): .is_empty() = false, .is_infinite() = true
+Range(-inf, 0, 10): .is_empty() = false, .is_infinite() = true
+Range(-inf, 0, inf): .is_empty() = false, .is_infinite() = true
+Range(inf, 0, nan): .is_empty() = false, .is_infinite() = true
+Range(0, inf, nan): .is_empty() = false, .is_infinite() = true
+Range(-inf, 0, nan): .is_empty() = false, .is_infinite() = true
+Range(0, -inf, nan): .is_empty() = false, .is_infinite() = true
+Range(nan, 0, inf): .is_empty() = false, .is_infinite() = true
+Range(0, nan, inf): .is_empty() = false, .is_infinite() = true
+Range(nan, 0, -inf): .is_empty() = false, .is_infinite() = true
+Range(0, nan, -inf): .is_empty() = false, .is_infinite() = true
+Range(inf, inf, 1): .is_empty() = false, .is_infinite() = true
+Range(inf, inf, -1): .is_empty() = false, .is_infinite() = true
+Range(inf, inf, 0): .is_empty() = false, .is_infinite() = true
+Range(inf, inf, nan): .is_empty() = false, .is_infinite() = true
+Range(inf, inf, inf): .is_empty() = false, .is_infinite() = true
+Range(inf, inf, -inf): .is_empty() = false, .is_infinite() = true
+Range(-inf, -inf, 1): .is_empty() = false, .is_infinite() = true
+Range(-inf, -inf, -1): .is_empty() = false, .is_infinite() = true
+Range(-inf, -inf, 0): .is_empty() = false, .is_infinite() = true
+Range(-inf, -inf, nan): .is_empty() = false, .is_infinite() = true
+Range(-inf, -inf, inf): .is_empty() = false, .is_infinite() = true
+Range(-inf, -inf, -inf): .is_empty() = false, .is_infinite() = true
+Range(nan, nan, 0): .is_empty() = false, .is_infinite() = true
+Range(nan, nan, 1): .is_empty() = false, .is_infinite() = true
+Range(nan, nan, inf): .is_empty() = false, .is_infinite() = true
+Range(nan, nan, -inf): .is_empty() = false, .is_infinite() = true
+```
+
+# How We Teach This
+
+This can be taught by improving the documentation of `Range` and by discussing a few examples in the tutorial section. Also, as stated above, `Range.is_infinite()` is hardly used on Github and the impact of this change should be considered mild.
+
+Tests can easily be added to `packages/collections/_test.pony` that would add a `_assert_empty` method and test for relevant cases while removing thoses cases from the current `_assert_infinite` tests.
+
+# Drawbacks
+
+* As far as I can tell from public code, Range.is_infinite() is not used in any Pony code other than test code associated with tests for Range.
+* This should not produce any bugs in the compiler since it is a small library change only
+* Maintenance cost should be marginal.
+
+# Alternatives
+
+The impact of not doing this are user code bugs in attempted uses of `Range` as shown in the motivation. At the very least the impact of loops over unintendedly infinite `Range` iterators should be minimized by either encouraging more use of `Range.is_infinite()` or by improved tutorials.
```

Comment:
Your belief that there will be user code bugs is editorialzing at this point. Alternatives should be presented with the best possible case for those alternatives such as "Leaving it as is means that code that might exist that I am unaware from my search on GitHub would continue to work".

Suggesting documentation or tutorial improvements if left as is is certainly reasonable to consider in this section.